costmap_2d/plugins/obstruction_layer.cpp

```diff
@@ -656,7 +721,11 @@ void ObstructionLayer::applyKernelAtLocation(std::shared_ptr<Kernel> kernel, flo
         unsigned int grid_idx = grid_x_start + grid_x;
         unsigned char grid_cost = grid[grid_idx];
         unsigned char kernel_cost = kernel->values_[kern_x_start + xx];
-        if (grid_cost < kernel_cost || grid_cost == NO_INFORMATION)
+
+        if (!(kernel->ignore_freespace_ && grid_cost == FREE_SPACE
```

Comment:
The intent here is not very clear, maybe separate out into bool values that describe the cases?  The compiler will optimize it into the same statement in the end but it will clarify the cases.